src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -329,7 +333,14 @@ public static string GetDefaultAddress(int pid)
 
             if (!TryGetDefaultAddress(pid, out string defaultAddress))
             {
-                throw new ServerNotAvailableException($"Process {pid} not running compatible .NET runtime.");
+                string msg = $"Process {pid} not running compatible .NET runtime.";
```

Comment:
We shouldn't assert the target process isn't running a compatible runtime up-front if we want reader to recognize there are other conditions that could also be the culprit.